connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/PluginDiscoveryMode.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.runtime.isolation;
+
+/**
+ * Method to use to discover plugins usable on a Connect worker.
```

Comment:
Nit: "Method" is a bit misleading (I thought this was incorrectly documented at first). Some synonyms from Google that could also work include "procedure", "technique", "process", and "mechanism".

Reply:
Also, related to the question in `WorkerConfig`--if the way we use this enum doesn't affect how plugins installed directly onto the classpath are discovered, perhaps we should say "isolated plugins" instead of just "plugins", and/or add a note about classpath-installed plugins being loaded no matter what?

Reply:
Also also, can't hurt to link the KIP here:
```suggestion
 * Method to use to discover plugins usable on a Connect worker.
 * @see <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-898%3A+Modernize+Connect+plugin+discovery">KIP-898</a>
```